minimesos/src/test/java/com/containersol/minimesos/HelloWorldContainer.java

```diff
@@ -28,8 +29,15 @@ protected void pullImage() {
 
     @Override
     protected CreateContainerCmd dockerCommand() {
-        return dockerClient.createContainerCmd(HELLO_WORLD_IMAGE).withPrivileged(true)
-                .withName( getName() );
+        ExposedPort exposedPort = ExposedPort.tcp(SERVICE_PORT);
+        Ports portBindings = new Ports();
+        portBindings.bind(exposedPort, Ports.Binding(SERVICE_PORT));
+        return dockerClient.createContainerCmd(HELLO_WORLD_IMAGE)
+                .withEnv(String.format("SERVICE_%d_NAME=%s", SERVICE_PORT, SERVICE_NAME))
+                .withPrivileged(true)
+                .withName(getName())
+                .withPortBindings(portBindings)
+                .withExposedPorts(exposedPort);
     }
 
```

Comment:
Yes it will and it will have a default name according to these specs: 
`<base(container-image)>[-<exposed-port> if >1 ports]`

> Using the base of the container image, if the image is gliderlabs/foobar, the service name is foobar. If the image is redis the service name is simply redis.

> Additionally, if a container has multiple exposed ports, it will append the internal exposed port to differentiate from each other. For example, an image nginx with two exposed ports, 80 and 443, will produce two services named nginx-80 and nginx-443.

> You can override this default name with label or environment variable SERVICE_NAME or SERVICE_x_NAME, where x is the internal exposed port. Note that if a container has multiple exposed ports then setting SERVICE_NAME will still result in multiple services named SERVICE_NAME-<exposed port>.

http://gliderlabs.com/registrator/latest/user/services/#service-name